operator/identity_gc.go

```diff
@@ -34,15 +35,35 @@ func startKvstoreIdentityGC() {
 	}
 	a := allocator.NewAllocatorForGC(backend)
 
+	successfulRuns := 0
+	failedRuns := 0
 	keysToDelete := map[string]uint64{}
 	go func() {
 		for {
-			keysToDelete2, err := a.RunGC(identityRateLimiter, keysToDelete)
+			keysToDelete2, gcStats, err := a.RunGC(identityRateLimiter, keysToDelete)
 			if err != nil {
 				log.WithError(err).Warning("Unable to run security identity garbage collector")
+
+				if operatorOption.Config.EnableMetrics {
+					failedRuns++
+					metrics.IdentityGCRuns.WithLabelValues(metrics.LabelValueOutcomeFail).Set(float64(failedRuns))
+				}
 			} else {
 				keysToDelete = keysToDelete2
+
+				if operatorOption.Config.EnableMetrics {
+					successfulRuns++
+					metrics.IdentityGCRuns.WithLabelValues(metrics.LabelValueOutcomeSuccess).Set(float64(successfulRuns))
+
+					if gcStats == nil {
+						log.Errorf("Update metric 'IdentityGCSize' failed: the returned gcStats is nil")
+					} else {
```

Comment:
Actually, is it possible for`gcStats` to be returned nil if there's no error? If not, then we don't need this check at all and we can avoid this error msg. The error msg is not actionable for the user anyway.

Reply:
`gcStats` should be non-nil if `err != nil`,  I removed the checking. Thanks!  @christarazi